app/autodiscovery/src/main/java/com/fsck/k9/autodiscovery/srvrecords/SrvServiceDiscovery.kt

```diff
@@ -0,0 +1,96 @@
+package com.fsck.k9.autodiscovery.srvrecords
+
+import com.fsck.k9.autodiscovery.ConnectionSettings
+import com.fsck.k9.autodiscovery.ConnectionSettingsDiscovery
+import com.fsck.k9.helper.EmailHelper
+import com.fsck.k9.mail.AuthType
+import com.fsck.k9.mail.ConnectionSecurity
+import com.fsck.k9.mail.ServerSettings
+import java.util.Locale
+import org.minidns.dnsname.DnsName
+import org.minidns.hla.ResolverApi
+import org.minidns.hla.SrvProto
+
+enum class SrvType(var protocol: String, var assumeTls: Boolean) {
+    SUBMISSION("smtp", false),
+    IMAPS("imap", true),
+    POP3S("pop", true),
+    IMAP("imap", false),
+    POP3("pop3", false)
+}
+
+data class MailService(
+    var srvType: SrvType,
+    var host: String,
+    var port: Int,
+    var priority: Int,
+    var security: ConnectionSecurity?
+)
+
+fun pickMailService(a: MailService?, b: MailService?): MailService? {
+    if (a == null || b == null) {
+        return a ?: b
+    } else {
+        return minOf(a, b, compareBy { it.priority })
+    }
+}
+
+class SrvServiceDiscovery : ConnectionSettingsDiscovery {
+
+    private var domain: String? = null
+
+    override fun discover(email: String): ConnectionSettings? {
+        this.domain = EmailHelper.getDomainFromEmailAddress(email)
+
+        val outgoingService = this.srvLookup(SrvType.SUBMISSION) ?: return null
+        val incomingService = pickMailService(
```

Comment:
Did you intend for this to only do lookups for imap and pop3 when neither imaps nor pop3s records were found? Wouldn't this produce an undesired result for e.g. this configuration:
```
_imap._tcp     SRV  0 1 143 imap.example.com.
_imaps._tcp    SRV  10 1 993 imap.example.com.
```